modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/NameClashModelValidator.java

```diff
@@ -213,6 +258,68 @@ private void validateClash(String existingNamingModel, String newNamingModel, St
                                                          typeOfNewNamingModel));
       }
     }
+
+    private void validateSingularizingClash(ParameterizedModel model) {
+      List<ParameterModel> parameters = model.getParameterModels();
+      parameters.forEach(
+                         p -> p.getType().accept(new MetadataTypeVisitor() {
+
+                           @Override
+                           public void visitArrayType(ArrayType arrayType) {
+                             validateSingular();
+                           }
+
+                           @Override
+                           public void visitDictionary(DictionaryType dictionaryType) {
+                             validateSingular();
+                           }
+
+                           private void validateSingular() {
+                             String singularName = singularize(p.getName());
+                             boolean allowInline = p.getModelProperty(XmlHintsModelProperty.class)
+                                 .map(XmlHintsModelProperty::allowsInlineDefinition)
+                                 .orElse(true);
+
+                             if (singularName.equals(p.getName()) || !allowInline) {
+                               return;
+                             }
+
+                             Optional<ParameterModel> clashingParam =
```

Comment:
all this, including the exception throwing => new method
validateNameClashWithParameters?